# 61011 Week 1

You should find the exercises in this level very simple to complete. All of the of the information required to complete these exercises can be found at www.learnpython.org. The cells below provide some skeleton code and some additional questions, but you should attempt to complete these exercises without using the skeleton code first.

If you **do** need to refer to the skeleton code, you will struggle with the remainder of the labs. In this case, we strongly advise going through the entire tutorial at www.learnpython.org before proceeding to the next notebook.

### Write a loop to sum the even numbers from 1 to 100
Before looking at the code below, think about whether a FOR loop or WHILE loop is more appropriate for this problem.

Type your answer here before revealing the answer below

Answer: Both loops can be used to complete the task. In general, a FOR loop should be used when the number of iterations (times around the loop) is pre-determined. This is the case here, when we know, before entering the loop, that we will loop through 50 even numbers

A WHILE loop is often slightly slower, as it needs to check a Boolean condition each time the loop is executed. Later in the course, we will see that WHILE loops are necessary when we have a stopping criterion for an iterative algorithm.

The code below is a skeleton. Replace the 'xx's 

In [5]:
count = xx
for i in range(xx, xx, xx):
    count = xx + xx
    
print(count)

2450


Answer: count = 2450

### sum the even numbers 1 to 100 without a loop

In [7]:
vec = range(xx,xx,xx)
count = sum(xx)
print(count)

2450


Answer: count = 2450

### Write a function to return the sum of its arguments and call it.

In [ ]:
def addXY(xx,xx):
    return xxxxxx
    
# call function below:
print(addXY(3,2))
print(addXY(1,-1))
print(addXY(500,30))

Answer: 5, 0, 530

### Write a function to return the sum of all the even numbers up to a given number, x

In [8]:
def sumEven(x):
    count = 0
    xxx
    xxx
    return(count)

print(sumEven(100))

IndentationError: expected an indented block (<ipython-input-8-aa0c272fce53>, line 2)